frontend/src/components/Question.vue

```diff
@@ -2,29 +2,24 @@
 import { computed, defineAsyncComponent, ref } from 'vue';
 import type { ComputedRef, Ref } from 'vue';
 import type { Quizoot } from '@interfaces/quizoot';
-import NavigationButton from '@/components/NavigationButton.vue';
+import type { Question } from '@interfaces/quizoot.indexed';
+import { useFetch } from '@/lib/hooks';
 import { pascalToSnake } from '@/lib/string-utils';
-
-interface QuestionsFlow {
-    questionNumber: number;
-    isFirstQuestion: boolean;
-    isLastQuestion: boolean;
-}
+import FetchError from '@/components/FetchError.vue';
+import Loader from '@/components/Loader.vue';
 
 interface QuestionWrapperProps {
```

Comment:
Now that I think of it, why are we repeating ourselves ? The component is already named `Question`. So the props should be : `id`, `nbQuestions` or `totalQuestion`, `pos` or `rank` (I don't really like `number`)

Reply:
You"re right. I chaange it like this:

```ts
// views/Quiz.vue
const currentQuestionRank: Ref<number> = ref(0);

// components/Question.vue
interface QuestionWrapperProps {
    id: Quizoot.Question['id'];
    rank: number;
    totalQuestions: number;
}
```